docs/sorting-algorithms/bubble-sort.mdx

```diff
@@ -0,0 +1,209 @@
+---
+title: Bubble Sort
+slug: /sorting-algorithms/bubble-sort
+---
+
+## Introduction
+
+Bubble sort is a simple sorting algorithm which compares two adjacent elements and swap if they are not in the correct position.
+
+## Explanation
+
+**AIM -** Repeatedly swap two adjacent elements if they are in the wrong order.
+
+**WORKING -** It works by iterating the array from the first element to the last element, comparing each pair of the elements, and swapping them if needed.
+
+First, we will take the size of the array from the user and will make an array of that size to store the elements given by the user. Now compare each pair of array elements unless the whole array is completely sorted in ascending order.
+
+Let's have one example:
+
+```
+Let say the user gives us the following elements: 5 4 3 2 1
+
+Bubble sort starts with the first two elements and checks which one is greater
+
+Pass 1.
+
+In this case value 5 is greater than 4, so it is swapped.
+| 5 | 4 | 3 | 2 | 1 |
+
+| 4 | 5 | 3 | 2 | 1 |
+
+Next we compare value 5 with 3 and this two value must be swapped.
+| 4 | 5 | 3 | 2 | 1 |
+
+| 4 | 3 | 5 | 2 | 1 |
+
+Next we compare value 5 with 2 and this also must be swapped.
+| 4 | 3 | 5 | 2 | 1 |
+
+| 4 | 3 | 2 | 5 | 1 |
+
+Then we move to the next two value which is 5 and 1, we swapped.
+| 4 | 3 | 2 | 5 | 1 |
+
+| 4 | 3 | 2 | 1 | 5 |
+
+Now 5 is bubbled.
+
+Notice that after each iteration, at least one value moves at the end.
+
+Likewise we are doing rest of passes to sort the array completely.
+Pass 2.
+
+| 4 | 3 | 2 | 1 |
+
+| 3 | 4 | 2 | 1 |
+
+| 3 | 2 | 4 | 1 |
+
+| 3 | 2 | 1 | 4 |
+
+Now 4 is bubbled.
+
+Pass 3.
+
+| 3 | 2 | 1 |
+
+| 2 | 3 | 1 |
+
+| 2 | 1 | 3 |
+
+Now 3 is bubbled.
+
+Pass 4.
+
+| 2 | 1 |
+
+| 1 | 2 |
+
+Now 2 is bubbled.
+
+So now our array now looks like::  1 2 3 4 5
+```
+
+## Algorithm
+
+- read total no. of elements say n.
+- store the elements in an array.
+- initially i = 0.
+- compare adjacent elements.
+- repeat step 4 for all n elements.
+- increment i by 1 and repeat steps 4,5 for i < n.
```

Comment:
Make starting letters capital, and use number instead of no.
Keep n, i, 4, 5 in backticks.

Reply:
done